src/exoticatechnologies/util/CircleUtils.kt

```diff
@@ -0,0 +1,820 @@
+             * @param particleColors depending on [particleDrawMode], it should either be a one-element list for [ParticleDrawMode.ONE_AT_A_TIME]
+             * or a list of colors for [ParticleDrawMode.WHOLE_ARM]
+             * @param particleBrightness particle brightness to use. Defaults to 1f
+             * @param particleDrawMode the mode in which to draw particles. See [ParticleDrawMode]
+             */
+            fun draw(
+                particleSize: Float = 12f,
+                particleDuration: Float = 0.25f,
+                particleColors: List<Color>,
+                particleBrightness: Float = 1f,
+                particleDrawMode: ParticleDrawMode
+            ) {
+                if (particlePoints.isNotEmpty()) {
+                    when(particleDrawMode) {
+                        ParticleDrawMode.ONE_AT_A_TIME -> {
+                            val point = particlePoints.removeAt(0)
+                            val engine = Global.getCombatEngine()
+                            // Fetch the original smooth particle limit
+//                            val originalLimit = (engine as CombatEngine).smoothParticles.limit
+                            // bump limit so they all fit
+//                            (engine as CombatEngine).smoothParticles.limit = particlePoints.size
+                            val particleColor = particleColors.first()
+                            EngineParticlePainter.addParticle(
+                                engine = engine,
+                                particleType = ParticleType.SMOOTH_PARTICLE,
+                                particleParams = ParticleParams.Smooth.Basic(
+                                    location = point,
+                                    velocity = Vector2f(0f, 0f),
+                                    size = particleSize,
+                                    brightness = particleBrightness,
+                                    duration = particleDuration,
+                                    color = particleColor
+                                )
+                            )
+                            // Revert limit after drawing
+//                            (engine as CombatEngine).smoothParticles.limit = originalLimit
+                        }
+                        ParticleDrawMode.WHOLE_ARM -> {
+                            // Fetch the original smooth particle limit
+                            val engine = Global.getCombatEngine()
+                            for (point in particlePoints) {
+                                val colorIndex = getCurrentPointIndex() / particleSegments
+                                val color = if (colorIndex < particleColors.size) {
+                                    particleColors[colorIndex]
+                                } else {
+                                    particleColors.last()
+                                }
```

Comment:
obviously this will break (crash) for an empty list. verify if that is covered somewhere in javadoc.